File: Source/Extensions/Tessa.Extensions.Default.Client/Documents/HideEmptyIncomingReferencesControl.cs
using System.Linq;
using System.Threading.Tasks;
using Tessa.UI.Cards;
using Tessa.UI.Cards.Controls.AutoComplete;

namespace Tessa.Extensions.Default.Client.Documents
{
    public sealed class HideEmptyIncomingReferencesControl : CardUIExtension
    {
        public override async Task Initialized(ICardUIExtensionContext context)
        {
            ICardModel model = context.Model;

            if (!model.Blocks.TryGet("RefsBlock", out var block))
            {
                return;
            }

            IControlViewModel control = block.Controls.FirstOrDefault(x => x.Name == "IncomingRefsControl");

            if (control is AutoCompleteTableViewModel autoComplete && autoComplete.Items.Count == 0)
            {
                control.ControlVisibility = System.Windows.Visibility.Collapsed;
            }
        }
    }
}

File: WebClient SDK/src/default/documents/hideEmptyIncomingReferencesControl.ts
import { CardUIExtension, ICardUIExtensionContext } from 'tessa/ui/cards';
import { AutoCompleteTableViewModel } from 'tessa/ui/cards/controls';
import { Visibility } from 'tessa/platform';

export class HideEmptyIncomingReferencesControl extends CardUIExtension {
  initialized(context: ICardUIExtensionContext): void {
    const model = context.model;

    const block = model.blocks.get('RefsBlock');
    if (!block) {
      return;
    }

    const control = block.controls.find(x => x.name === 'IncomingRefsControl');
    const autoComplete = control as AutoCompleteTableViewModel;

    if (
      autoComplete &&
      autoComplete instanceof AutoCompleteTableViewModel &&
      autoComplete.items.length === 0
    ) {
      autoComplete.controlVisibility = Visibility.Collapsed;
    }
  }
}
